src/Storage/Adapter/Redis.php

```diff
@@ -195,18 +197,34 @@ function ($value) {
     }
 
     /**
-     * Add the specified values to the set stored at key.
+     * Get tags of an item by given key
      *
-     * @param  string $normalizedKey
-     * @param  mixed  $value
-     * @return bool
+     * @param string $key
+     * @return string[]|FALSE
+     */
+    public function getTags($key)
+    {
+        $redis = $this->getRedisResource();
+        try {
+            return $redis->sMembers($this->namespacePrefix . $key);
```

Comment:
I dont think that this is what you want. This method is to provide all tags the given key is in. What you implemented is, get all members of given tag.

I tried to implement something like this aswell, but since redis itself does not provide any logic to grab all tags for a given key, I tried something like this (which is - imho - quite expensive):

    /**
     * {@inheritdoc}
     *
     * WARNING: This operation is very heavy since redis does not provide native logic for this! Use with caution!
     */
    public function getTags($key)
    {
        $this->normalizeKey($key);

        if (! $this->internalHasItem($key)) {
            return false;
        }

        $resource = $this->getRedisResource();

        $availableSets = $this->getKeysByType(static::TYPE_SET);
        $sets = [];
        foreach ($availableSets as $set) {
            $members = $resource->sMembers($set);
            if (in_array($key, $members)) {
                array_push($sets, $set);
            }
        }

        return $sets;
    }

    /**
     * Returns all keys by given type
     *
     * @param string $type
     *
     * @return array
     *
     * @throws InvalidArgumentException If given type is not string of not in allowed types.
     */
    protected function getKeysByType($type)
    {
        if (! is_string($type)) {
            throw new InvalidArgumentException(sprintf(
                'Given type must be string; %s given',
                is_object($type) ? get_class($type) : gettype($type)
            ));
        }

        $allowedTypes = [
            static::TYPE_SET,
            static::TYPE_ZSET,
            static::TYPE_LIST,
            static::TYPE_STRING,
        ];

        if (! in_array($this, $allowedTypes)) {
            throw new InvalidArgumentException(sprintf(
                'Given type must be one of "%s"; "%s" given',
                implode('", "', $allowedTypes),
                $type
            ));
        }

        $resource = $this->getRedisResource();

        $keys = $resource->keys('*');
        $keysMatchingType = [];

        foreach ($keys as $key) {
            if ($resource->type($key) === $type) {
                array_push($keysMatchingType, $type);
            }
        }

        return $keysMatchingType;
    }